### ISSUE TITLE
Incorrectly highlighted regex

### ISSUE THREAD

**Issue by the-pirate-bay** [codemirror/CodeMirror#4981]
### Issue

In some conditions a regex is highlighted like a comment and vice versa. The issue is originally filed at Chromium project. Relevant issues are [this](https://bugs.chromium.org/p/chromium/issues/detail?id=724586), [this](https://bugs.chromium.org/p/chromium/issues/detail?id=725361), [this](https://bugs.chromium.org/p/chromium/issues/detail?id=725466), [this](https://bugs.chromium.org/p/chromium/issues/detail?id=725482) and [this](https://bugs.chromium.org/p/chromium/issues/detail?id=725484). Almost all issues have been merged into the last one due to similarity.

### Test cases

The highlighter fails for the following test cases:

```JavaScript
[.../\//];

void/\//;

do/\//
while(0);

if(0);else/\//;

`${async++}//`;
`${await++}//`;
`${let++}//`;

+{1:+{}/undefined};

`${{}/ /\//}`;

+class extends{1:{}/ /\//}{};

(a={}/ /\//)=>a;

{}/./gym?
{}/ /\//:
{}/ /\//;
```

### Explanation

Each of the test cases represents a valid JavaScript code according to the latest ES standard specification. However, such examples, although they are valid, they will never appear in a serious script, until someone's intent is to craft a programming puzzle. For example, the first test case is useless until someone implements **RegExp.prototype[Symbol.iterator]**. Another example (the second test case) makes no sense, because it is equivalent to **undefined** and there's no need for a regex at all.

Therefore, if fixing these issues will yield a huge impact on the performance, I strongly recommend not to spend time on fixing it. It would be good to cover all test cases, but if it would require major changes, impact performance or cause complexity problems, then probably it would be better not to bother with this at all.

I've just reported this because the members of Chromium project have [suggested](https://i.stack.imgur.com/lwwtx.png) it, but I'm aware that the fix may probably have a big impact on the performance. So, you will decide if it is worth fixing the issue or not.

**Comment by marijnh** [comment]
Are these the output of a  fuzzer?

Most of them weren't too hard to fix, and are fixed in 036beb54. Distinguishing between keyword `let`/`await` and regular identifiers by that name isn't something this mode does yet, since it seems the world is moving on from using them as identifiers. It's probably also not too hard, but it seems like it's not worth it.

**Comment by the-pirate-bay** [comment]
Thanks for the fix. As answer to your question, no fuzzer is used. I found the examples manually.
